XenModel/Actions/HealthCheck/GetHealthCheckAnalysisResultAction.cs

```diff
@@ -0,0 +1,154 @@
+                if (!Pool.HealthCheckSettings.HasUpload)
+                {
+                    log.InfoFormat("Cannot get the diagnostic result for {0}, because the there is no upload completed yet", Pool.Name);
+                    Description = Messages.ACTION_GET_HEALTH_CHECK_RESULT_FAILED;
+                    return;
+                }
+                var analysisResult = GetAnalysisResult(diagnosticToken, Pool.HealthCheckSettings.UploadUuid);
+                log.Info("Saving analysis result");
+                Dictionary<string, string> newConfig = Pool.health_check_config;
+                newConfig[HealthCheckSettings.REPORT_ANALYSIS_SEVERITY] = HealthCheckSettings.DiagnosticAlertSeverityToString(GetMaxSeverity(analysisResult));
+                newConfig[HealthCheckSettings.REPORT_ANALYSIS_ISSUES_DETECTED] = analysisResult.Count.ToString();
+                newConfig[HealthCheckSettings.REPORT_ANALYSIS_UPLOAD_UUID] = Pool.HealthCheckSettings.UploadUuid;
+                newConfig[HealthCheckSettings.REPORT_ANALYSIS_UPLOAD_TIME] = Pool.HealthCheckSettings.LastSuccessfulUpload;
+                Pool.set_health_check_config(Session, Pool.opaque_ref, newConfig);
+            }
+            catch (Exception e)
+            {
+                log.InfoFormat("Exception while getting diagnostic result from {0}. Exception Message: {1} ", diagnosticDomainName, e.Message);
+                Description = Messages.ACTION_GET_HEALTH_CHECK_RESULT_FAILED;
+                throw;
+            }
+            
+            Description = Messages.COMPLETED;
+        }
+
+        private DiagnosticAlertSeverity GetMaxSeverity(List<AnalysisResult> diagnosticAlerts)
+        {
+            var maxSeverity = DiagnosticAlertSeverity.Info;
+            foreach (var diagnosticAlert in diagnosticAlerts)
+            {
+                var severity = HealthCheckSettings.StringToDiagnosticAlertSeverity(diagnosticAlert.severity);
+                if (severity > maxSeverity)
+                    maxSeverity = severity;
+            }
+            return maxSeverity;
+        }
+
+        private List<AnalysisResult> GetAnalysisResult(string diagnosticToken, string uploadUuid)
+        {
+            var urlString = string.Format("{0}{1}?upload_uuid={2}", diagnosticDomainName, DIAG_RESULT_URL, uploadUuid);
+            var authorizationHeader = "BT " + diagnosticToken;
+            var httpWebRequest = (HttpWebRequest)WebRequest.Create(urlString);
+            httpWebRequest.Headers.Add("Authorization", authorizationHeader);
+            httpWebRequest.Method = "GET";
+
+            string result;
+            var httpResponse = (HttpWebResponse)httpWebRequest.GetResponse();
+            using (var streamReader = new StreamReader(httpResponse.GetResponseStream()))
+            {
+                result = streamReader.ReadToEnd();
+            }
+            TemplateResponse response = new JavaScriptSerializer().Deserialize<TemplateResponse>(result);
```

Comment:
But this is an action, so it is fine. Did not think it before 
:walking: to merge